server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -509,6 +551,14 @@ public void validate(IndexMode value, Map<Setting<?>, Object> settings) {
      * The {@link IndexMode "mode"} of the index.
      */
     private final IndexMode mode;
+    /**
+     * Start time of the time_series index.
+     */
+    private final Instant timeSeriesStartTime;
+    /**
+     * End time of the time_series index.
+     */
+    private volatile Instant timeSeriesEndTime;
```

Comment:
I'm worried that right now these can't be `null` so all indices are going to validate the range. I wonder if we can convert these to `null` for the defaults so we disable the validation when it isn't set.